server/src/main/java/org/elasticsearch/cluster/metadata/DataStream.java

```diff
@@ -171,6 +175,32 @@ public Index getWriteIndex() {
         return indices.get(indices.size() - 1);
     }
 
+    public Index selectWriteIndex(Instant timestamp, Metadata metadata) {
```

Comment:
Can you rename this to be something TSDB specific, like `selectTimeseriesWriteIndex` (since we may end up adding different selection criteria in the future) and add javadocs?

Reply:
Renamed and add jdocs: 36af27e2bdfaa9d58d9647265574e9754c8fd373